src/platform/export/exportBase.node.ts

```diff
@@ -1,46 +1,83 @@
 import { inject, injectable } from 'inversify';
 import * as path from '../../platform/vscode-path/path';
-import { CancellationToken, Uri } from 'vscode';
+import { CancellationToken, NotebookDocument, Uri } from 'vscode';
 import { INotebookImporter } from '../../kernels/jupyter/types';
 import { IJupyterSubCommandExecutionService } from '../../kernels/jupyter/types.node';
 import { IFileSystemNode } from '../common/platform/types.node';
 import { IPythonExecutionFactory, IPythonExecutionService } from '../common/process/types.node';
 
-import { reportAction } from '../progress/decorator.node';
+import { reportAction } from '../progress/decorator';
 import { ReportableAction } from '../progress/types';
 import { PythonEnvironment } from '../pythonEnvironments/info';
-import { ExportFormat, INbConvertExport } from './types';
+import { ExportFormat, IExportBase, IExportDialog, INbConvertExport } from './types';
+import { ExportUtil } from './exportUtil.node';
+import { TemporaryDirectory } from '../common/platform/types';
+import { ExportInterpreterFinder } from './exportInterpreterFinder.node';
 
 @injectable()
-export class ExportBase implements INbConvertExport {
+export class ExportBase implements INbConvertExport, IExportBase {
     constructor(
         @inject(IPythonExecutionFactory) protected readonly pythonExecutionFactory: IPythonExecutionFactory,
         @inject(IJupyterSubCommandExecutionService)
         protected jupyterService: IJupyterSubCommandExecutionService,
         @inject(IFileSystemNode) protected readonly fs: IFileSystemNode,
-        @inject(INotebookImporter) protected readonly importer: INotebookImporter
+        @inject(IExportDialog) protected readonly filePicker: IExportDialog,
+        @inject(ExportUtil) protected readonly exportUtil: ExportUtil,
+        @inject(INotebookImporter) protected readonly importer: INotebookImporter,
+        @inject(ExportInterpreterFinder) private exportInterpreterFinder: ExportInterpreterFinder
     ) {}
 
     public async export(
```

Comment:
is this method needed in these classes?

Reply:
I still put it there since we register it as `serviceManager.addSingleton<INbConvertExport>(INbConvertExport, ExportBase, 'Export Base');`, not sure why it's needed there. I can send a pr later to have it fixed.